src/main/java/io/kamax/matrix/json/event/MatrixJsonRoomPowerLevelsEvent.java

```diff
@@ -0,0 +1,199 @@
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package io.kamax.matrix.json.event;
+
+import com.google.gson.JsonObject;
+
+import io.kamax.matrix.event._RoomPowerLevelsEvent;
+import io.kamax.matrix.json.MatrixJsonObject;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class MatrixJsonRoomPowerLevelsEvent extends MatrixJsonRoomEvent implements _RoomPowerLevelsEvent {
+
+    private Content content;
+
+    public MatrixJsonRoomPowerLevelsEvent(JsonObject obj) {
+        super(obj);
+
+        content = new Content(getObj("content"));
+    }
+
+    @Override
+    public Optional<Double> getBan() {
+        return Optional.ofNullable(content.getBan());
+    }
+
+    @Override
+    public Map<String, Double> getEvents() {
+        return Collections.unmodifiableMap(content.getEvents());
+    }
+
+    @Override
+    public Optional<Double> getEventsDefault() {
+        return Optional.ofNullable(content.getEventsDefault());
+    }
+
+    @Override
+    public Optional<Double> getInvite() {
+        return Optional.ofNullable(content.getInvite());
+    }
+
+    @Override
+    public Optional<Double> getKick() {
+        return Optional.ofNullable(content.getKick());
+    }
+
+    @Override
+    public Optional<Double> getRedact() {
+        return Optional.ofNullable(content.getRedact());
+    }
+
+    @Override
+    public Optional<Double> getStateDefault() {
+        return Optional.ofNullable(content.getStateDefault());
+    }
+
+    @Override
+    public Map<String, Double> getUsers() {
+        return Collections.unmodifiableMap(content.getUsers());
+    }
+
+    @Override
+    public Optional<Double> getUsersDefault() {
+        return Optional.ofNullable(content.getUsersDefault());
+    }
+
+    private class Content extends MatrixJsonObject {
+
+        private Double ban;
+        private Map<String, Double> events = new HashMap<>();
+        private Double eventsDefault;
+        private Double invite;
+        private Double kick;
+        private Double redact;
+        private Double stateDefault;
+        private Map<String, Double> users = new HashMap<>();
+        private Double usersDefault;
+
+        Content(JsonObject obj) {
+            super(obj);
+
+            setBan(getDoubleIfPresent("ban"));
+            setEventsDefault(getDoubleIfPresent("events_default"));
+            setInvite(getDoubleIfPresent("invite"));
+            setKick(getDoubleIfPresent("kick"));
+            setRedact(getDoubleIfPresent("redact"));
+            setStateDefault(getDoubleIfPresent("state_default"));
+            setUsersDefault(getDoubleIfPresent("users_default"));
+
+            JsonObject eventsJson = obj.getAsJsonObject("events");
+            JsonObject usersJson = obj.getAsJsonObject("users");
+
+            Map<String, Double> eventsMap = eventsJson.entrySet().stream()
```

Comment:
If the `events` key does not exist, wouldn't this be null and fail on `.entrySet()` ?